crates/ironrdp-pdu/src/rdp/capability_sets/general.rs

```diff
@@ -97,22 +97,33 @@ bitflags! {
 pub struct General {
     pub major_platform_type: MajorPlatformType,
     pub minor_platform_type: MinorPlatformType,
+    pub protocol_version: u16,
     pub extra_flags: GeneralExtraFlags,
     pub refresh_rect_support: bool,
     pub suppress_output_support: bool,
 }
 
+impl Default for General {
+    fn default() -> Self {
+        Self {
+            major_platform_type: MajorPlatformType::UNSPECIFIED,
+            minor_platform_type: MinorPlatformType::UNSPECIFIED,
+            protocol_version: PROTOCOL_VER,
+            extra_flags: GeneralExtraFlags::empty(),
+            refresh_rect_support: Default::default(),
+            suppress_output_support: Default::default(),
+        }
+    }
+}
+
 impl PduParsing for General {
     type Error = CapabilitySetsError;
 
     fn from_buffer(mut buffer: impl io::Read) -> Result<Self, Self::Error> {
         let major_platform_type = MajorPlatformType(buffer.read_u16::<LittleEndian>()?);
         let minor_platform_type = MinorPlatformType(buffer.read_u16::<LittleEndian>()?);
 
-        let protocol_ver = buffer.read_u16::<LittleEndian>()?;
-        if protocol_ver != PROTOCOL_VER {
-            return Err(CapabilitySetsError::InvalidProtocolVersion);
-        }
```

Comment:
To be a bit more conservative, do you think you could add a warning when `protocol_ver` is different from 0x0200 at this place?

https://github.com/Devolutions/IronRDP/blob/78caad219a9c7f701c016b8dc45a6f94c46626d0/crates/ironrdp-connector/src/connection.rs#L706

We may want a similar check in the acceptor as well, so we when a version we didn’t originally expect is received

Reply:
@CBenoit there is no issue adding "tracing" as dependency of "ironrdp-pdu"? ok.

Reply:
oh ok!